# `no-extra-parens`

<<<<<<< HEAD
Disallows unnecessary parentheses.
=======
Disallow unnecessary parentheses.
>>>>>>> fe

## Rule Details

This rule extends the base [`eslint/no-extra-parens`](https://eslint.org/docs/rules/no-extra-parens) rule.
It adds support for TypeScript type assertions.

## How to Use

```jsonc
{
  // note you must disable the base rule as it can report incorrect errors
  "no-extra-parens": "off",
  "@typescript-eslint/no-extra-parens": ["error"]
}
```

## Options

See [`eslint/no-extra-parens` options](https://eslint.org/docs/rules/no-extra-parens#options).

<sup>

Taken with ‚ù§Ô∏è [from ESLint core](https://github.com/eslint/eslint/blob/main/docs/rules/no-extra-parens.md)

</sup>
<<<<<<< HEAD
=======

## Attributes

- Configs:
  - [ ] ‚úÖ Recommended
  - [ ] üîí Strict
- [x] üîß Fixable
- [ ] üí≠ Requires type information
>>>>>>> fe
