# `no-misused-new`

<<<<<<< HEAD
Enforces valid definition of `new` and `constructor`.
=======
Enforce valid definition of `new` and `constructor`.
>>>>>>> fe

Warns on apparent attempts to define constructors for interfaces or `new` for classes.

## Rule Details

Examples of code for this rule:

<!--tabs-->

### ‚ùå Incorrect

```ts
class C {
  new(): C;
}

interface I {
  new (): I;
  constructor(): void;
}
```

### ‚úÖ Correct

```ts
class C {
  constructor() {}
}
interface I {
  new (): C;
}
```

## Options

```jsonc
// .eslintrc.json
{
  "rules": {
    "@typescript-eslint/no-misused-new": "error"
  }
}
```

This rule is not configurable.
<<<<<<< HEAD
=======

## Related To

- TSLint: [no-misused-new](https://palantir.github.io/tslint/rules/no-misused-new/)

## Attributes

- Configs:
  - [x] ‚úÖ Recommended
  - [x] üîí Strict
- [ ] üîß Fixable
- [ ] üí≠ Requires type information
>>>>>>> fe
